gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -1947,3 +1929,91 @@ def test_unsupported_parameter_rest_asyncio():
     )
 
 {% endmacro %}
+
+{# is_rest_unsupported_method renders:
+ # 'True' if transport is async REST.
+ # 'True' if transport is sync REST and method is a client streaming method.
+ # 'False' otherwise.
+#}
+{# NOTE: We will keep updating this method as we add support for methods in async REST. #}
+{% macro is_rest_unsupported_method(method, is_async) %}
+{%- if method.client_streaming or is_async -%}
+{{'True'}}
+{%- else -%}
+{{'False'}}
+{%- endif -%}
+{% endmacro %}
+
+{# rest_transport_specific_tests generates all the rest specific tests for both
+# sync and async transport. It refactors the test cases in macro::rest_required_tests
+# and should replace it once we cover all the test cases in it here.
+#}
+{% macro rest_transport_specific_tests(service, transport, is_async) %}
+{% if 'rest' in transport %}
+{### Test syntax configuration begins ###}
+{% set await_prefix = "await " if is_async else "" %}
+{% set async_prefix = "async " if is_async else "" %}
+{% set async_decorator = "@pytest.mark.asyncio " if is_async else "" %}
+{% set transport_name = get_transport_name(transport, is_async) %}
+{### Test syntax configuration ends ###}
+{### Test generation for all methods begins ###}
+{% for method in service.methods.values() %}
+{% set method_name = method.name|snake_case %}
+{### test cases section begin ###}
+{% if is_rest_unsupported_method(method, is_async) == 'True' or not method.http_options %}
+{{ rest_method_not_implemented_error(service, method, method_name, transport_name, async_prefix, await_prefix, async_decorator, is_async) }}
+{% endif %}{# is_rest_unsupported_method(method, is_async) == 'False' and method.http_options #}
+{### test cases section ends ###}
+{% endfor %}
+{### Test generation for all methods ends ###}
+{{ rest_method_initialize_client_test(service, transport_name, is_async) }}
```

Comment:
I've been following an approach where I add `_test` as a suffix to the macro name and `test_` as a prefix to the generated test name. Let me know if you find any inconsistency in the changes introduced and I'm happy to fix it!

I agree that we should be consistent with the naming!

Reply:
OK, thanks for having a policy!

 I was also referring to consistency with pre-existing macros, even before this PR. Making macro naming consistent overall would be a nice cosmetic clean-up, but obviously not critical. We can follow up at some point (though honestly, it seems like this will be near the bottom of our list for a good time, since it's not that important).